src/executor/admin/SpaceExecutor.cpp

```diff
@@ -10,24 +10,62 @@
 #include "planner/Admin.h"
 #include "util/SchemaUtil.h"
 #include "util/ScopedTimer.h"
+#include "common/plugin/fulltext/elasticsearch/ESGraphAdapter.h"
 
 namespace nebula {
 namespace graph {
 folly::Future<Status> CreateSpaceExecutor::execute() {
     SCOPED_TIMER(&execTime_);
 
     auto *csNode = asNode<CreateSpace>(node());
-    return qctx()->getMetaClient()->createSpace(csNode->getSpaceDesc(), csNode->getIfNotExists())
+    meta::MetaClient *client = qctx()->getMetaClient();
+    return client->createSpace(csNode->getSpaceDesc(), csNode->getIfNotExists())
             .via(runner())
-            .then([](StatusOr<bool> resp) {
+            .then([this, csNode, client](StatusOr<bool> resp) {
                 if (!resp.ok()) {
                     LOG(ERROR) << resp.status();
                     return resp.status();
                 }
+                if (csNode->getSpaceDesc().text_search) {
+                    auto ret = client->getFTClientsFromCache();
+                    if (!ret.ok() || ret.value().empty()) {
+                        return Status::Error("text search clients not found");
+                    }
+                    auto spaceName = csNode->getSpaceDesc().space_name;
+                    auto edgeTSIndex = nebula::plugin::IndexTraits::indexName(spaceName, true);
+                    if (createTSIndex(ret.value(), edgeTSIndex)) {
+                        return Status::Error("external index create failed : %s",
+                                             edgeTSIndex.c_str());
+                    }
+
+                    auto tagTSIndex = nebula::plugin::IndexTraits::indexName(spaceName, false);
+                    if (createTSIndex(ret.value(), tagTSIndex)) {
+                        return Status::Error("external index create failed : %s",
+                                             tagTSIndex.c_str());
+                    }
+                }
```

Comment:
I am not pretty sure we should create index here. Why we don't `CREATE FULLTEXT TAG | EDGE INDEX index_name ON space_name [fulltext_clause]` like in design doc?

Reply:
> I am not pretty sure we should create index here. Why we don't `CREATE FULLTEXT TAG | EDGE INDEX index_name ON space_name [fulltext_clause]` like in design doc?

Good question. the original design is similar to what you said.after an offline discussion with Sherman. I modified it to create external indexes and paces together. For two reasons: 1, external indexes are associated with space, not with tag or edge; 2, external index creation does not require meta; the graph layer itself is ok.